build.zig

```diff
@@ -94,6 +97,17 @@ pub const ScanProtocolsStep = struct {
         ) catch unreachable);
     }
 
+    pub fn interfaces(self: *ScanProtocolsStep, ifaces: []const scanner.RequiredInterface) void {
+        for (ifaces) |iface| {
+            // wl_display, wl_registry, and wl_callback are always generated
+            // as their version cannot change.
+            if (mem.eql(u8, iface.name, "wl_display") or mem.eql(u8, iface.name, "wl_registry") or
+                mem.eql(u8, iface.name, "wl_callback")) continue;
```

Comment:
We should print an error message here and panic instead of ignoring these, also this logic belongs in `scanner.zig` IMO.